frontend/lib/resolution/prims.cpp

```diff
@@ -91,10 +251,67 @@ CallResolutionResult resolvePrimCall(Context* context,
     case PRIM_IS_CONST_ASSIGNABLE:
     case PRIM_HAS_DEFAULT_VALUE:
     case PRIM_NEEDS_AUTO_DESTROY:
+
     case PRIM_CALL_RESOLVES:
-    case PRIM_METHOD_CALL_RESOLVES:
     case PRIM_CALL_AND_FN_RESOLVES:
     case PRIM_METHOD_CALL_AND_FN_RESOLVES:
+    case PRIM_METHOD_CALL_RESOLVES: {
+        bool forMethod = prim == PRIM_METHOD_CALL_RESOLVES ||
+                         prim == PRIM_METHOD_CALL_AND_FN_RESOLVES;
+        bool resolveFn = prim == PRIM_CALL_AND_FN_RESOLVES ||
+                         prim == PRIM_METHOD_CALL_AND_FN_RESOLVES;
+
+        if ((forMethod && ci.numActuals() < 2) ||
+            (!forMethod && ci.numActuals() < 1)) break;
+
+        size_t fnNameActual = forMethod ? 1 : 0;
+
+        UniqueString fnName;
+        if (toParamStringActual(ci.actual(fnNameActual).type(), fnName)) {
+          bool callAndFnResolved = false;
+
+          std::vector<CallInfoActual> actuals;
+          if (forMethod) {
+            actuals.push_back(CallInfoActual(ci.actual(0).type(), USTR("this")));
+          }
+          for (size_t i = fnNameActual + 1; i < ci.numActuals(); i++) {
+            actuals.push_back(ci.actual(i));
+          }
+          auto callInfo = CallInfo(fnName,
+                                   /* calledType */ QualifiedType(),
+                                   /* isMethodCall */ forMethod,
+                                   /* hasQuestionArg */ false,
+                                   /* isParenless */ false,
+                                   std::move(actuals));
+          auto resolvedCall = resolveGeneratedCall(context, call, callInfo,
```

Comment:
It might not be the case today, but I would expect `resolveGeneratedCall` to emit errors in some situations. As a result, I think this code should use `runAndTrackErrors` for `resolveGeneratedCall` here as well.